spec/etcdv3_spec.rb

```diff
@@ -381,6 +381,38 @@
           end
         end
       end
+
+      describe 'txn.lease' do
+        let!(:lease_id) { conn.lease_grant(2)['ID'] }
+        before { conn.put('txn', 'value') }
+        after { conn.del('txn') }
+        context 'success' do
+          subject! do
+            conn.transaction do |txn|
+              txn.compare = [ txn.lease('txn', :equal, lease_id) ]
+              txn.success = [ txn.put('txn-test', 'success') ]
+              txn.failure = [ txn.put('txn-test', 'failed') ]
+            end
+          end
+          it 'sets correct key' do
+            expect(conn.get('txn-test').kvs.first.value).to eq('success')
+          end
+        end
+        context 'failure' do
+          subject! do
+            conn.transaction do |txn|
+              txn.compare = [ txn.lease('txn', :equal, 123456)]
+              txn.success = [ txn.put('txn-test', 'success') ]
+              txn.failure = [ txn.put('txn-test', 'failed') ]
+            end
+          end
+          it 'sets correct key' do
+            puts conn.get('txn').inspect
+            expect(conn.get('txn-test').kvs.first.value).to eq('failed')
```

Comment:
For some reason the txn.lease comparisons return successful no matter what the comparison is.

Reply:
Apparently this hasn't been put into a release yet... Holding off on this..